search.py

```diff
@@ -477,7 +477,8 @@ def genetic_algorithm(population, fitness_fn, ngen=1000, pmut=0.1):
     "[Fig. 4.8]"
     for i in range(ngen):
         new_population = []
-        for i in len(population):
+        len_population = len(population)
+        for i in len_population:
```

Comment:
@anuj1995 Do you mean the ngen loop or the population loop? I was talking about the population loop. If we change it to range(len_population), evaluation of len(population) should be done before the `for i in range(ngen)` loop(assuming that the population size won't change through the generations)